### ISSUE TITLE
RIP loss in cs_gan repo

### ISSUE THREAD

**Issue by junja94** [deepmind/deepmind-research#155]
Hi, 

It seems like the implementation of the Restricted Isometry Property (RIP) loss is slightly different from the formulation.
Link to code: https://github.com/deepmind/deepmind-research/blob/1fcac77dc8697fa6c08cba39d2e91bd3c0317b0b/cs_gan/cs.py#L101

According to the paper and the comment,
RIP loss = (\sqrt{F(x_1 - x_2)^2} - \sqrt{(x_1 - x_2)^2})^2 .

As far as I understood, the first error term (F(x_1 - x_2)) should be implemented as:
  self.measure (img1 - img2) 

instead of 
    m1 = self._measure(img1)
    m2 = self._measure(img2)
    m2 - m1
, which is in the code.

If the measurement function is a linear map, both formulations should be identical.
I wonder if it is the same case for a neural network measurement function.
